nyc_data/ppe/data_import.py

```diff
@@ -0,0 +1,80 @@
+import collections
+import dataclasses as dc
+import hashlib
+from pathlib import Path
+from typing import Optional
+
+from ppe.data_mappings import DataSource
+from ppe import data_mappings
+from ppe.models import Purchase, Delivery, Inventory, ImportStatus, DataImport
+from xlsx_utils import import_xlsx
+
+mappings = {
+    DataSource.EDC_MAKE: data_mappings.SUPPLIERS_AND_PARTNERS,
+    DataSource.INVENTORY: data_mappings.INVENTORY,
+    DataSource.EDC_PPE: data_mappings.DCAS_DAILY_SOURCING
+}
+
+
+class DataImportError(Exception):
+    pass
+
+
+def import_in_progress(data_source: DataSource):
+    return DataImport.objects.filter(data_source=data_source, status=ImportStatus.candidate)
+
+
+def import_data(path: Path, data_source: DataSource, uploaded_by: Optional[str] = None, overwrite_in_prog=False):
+    in_progress = import_in_progress(data_source)
+
+    if in_progress.count() > 0:
+        if overwrite_in_prog:
+            in_progress.update(status=ImportStatus.replaced)
+        else:
+            raise DataImportError(
+                f"Can't import data -- there is already an import in progress ({in_progress.first().import_date})")
+
+    with open(path, 'rb') as f:
+        checksum = hashlib.sha256(f.read())
```

Comment:
Just to make sure I'm tracking, this `hashlib.sha256()` is to a step to preserve security, correct? So we're storing the hashed version of the data that is being imported, as opposed to the plaintext version?

Reply:
some of the files are >1MB and we're on a tiny free DB. We could also store the entire file, I don't think there's really a security worry here. This was to facilitate seeing if we've uploaded the exact same file before although I didn't code it

Reply:
Ahh got it, OK thanks. Hash functions take a variable length input, and outputs a fixed length hash value. Which, to your point, allows for more predictable retrieval times in the DB. According to link below, hashing is "a computationally and storage space efficient form of data access which avoids...the often exponential storage requirements of direct access of...large or variable-length keys" (second paragraph)

https://en.wikipedia.org/wiki/Hash_function